File: Flyweight/Golang/CounterTerrorist.go
package main

import "fmt"

type CounterTerrorist struct {
	task string
}

func (counterTerrorist *CounterTerrorist) assignTask(task string)  {
	counterTerrorist.task = task
}

func (counterTerrorist *CounterTerrorist) doMission()  {
	fmt.Println("Counter Terrorist doing Task :", counterTerrorist.task)
}

File: Flyweight/PHP/CounterTerrorist.php
<?php

require_once "IPlayer.php";

class CounterTerrorist implements IPlayer {
    private string $task;

    public function assignTask(string $task) {
        $this->task = $task;
    }

    public function doMission()
    {
        echo "Counter Terrorist doing Task : " . $this->task . "\n";
    }
}